objcgen/processor.cs

```diff
@@ -67,7 +67,7 @@ public void Process (ProcessedAssembly a)
 
 		protected void AddExtraType (ProcessedType pt)
 		{
-			typeQueue.Enqueue (pt);
+			Types.Add (pt);
```

Comment:
That does not work, e.g. when `AddExtraType` is called while processing the `typeQueue`. `Types` was cleared and won't be used again while dequeuing.

I _think_ what's needed to create an empty queue and add `Types` to it (then clear it). What was the test case ? just run on mscorlib.dll ?